types.bal

```diff
@@ -207,14 +193,23 @@ public type TimelineStepListResponse record {|
     int itemCount;
 |};
 
+public type TimelineSubstepPost record {|
+    string resultLocation;
+    string[] inputData;
+    string? parameters;
+    *database:Progress;
```

Comment:
Substeps do not need any progress as progress is handled by the step

Reply:
The idea was to encapsulate new progress data with input data if the UI posts new input data for a substep in `post experiments/[int experimentId]/timeline/[int timelineStep]/substeps/[int substepNr](@http:Payload TimelineSubstepPost substepData)`. Not sure how else this shoud be done or maybe not update progress in this post?